src/lib/logger/logger.ts

```diff
@@ -3,39 +3,42 @@ import { isValidObject } from '../middleware/utils';
 import { Level } from './level';
 import RecordFactory from './recordFactory';
 import RecordWriter from './recordWriter';
-import RequestContext from './requestContext';
+import Context from './context';
 
 export default class Logger {
     private parent?: Logger = undefined
-    private context?: RequestContext;
+    private context?: Context;
     private registeredCustomFields: Array<string> = [];
     private customFields: Map<string, any> = new Map<string, any>()
     private recordFactory: RecordFactory;
     private recordWriter: RecordWriter;
     protected loggingLevelThreshold: Level = Level.Inherit
 
-    constructor(parent?: Logger, reqContext?: RequestContext) {
+    constructor(parent?: Logger, context?: Context) {
         if (parent) {
             this.parent = parent;
             this.registeredCustomFields = parent.registeredCustomFields;
         }
-        if (reqContext) {
-            this.context = reqContext;
+        if (context) {
+            this.context = context;
         }
         this.recordFactory = RecordFactory.getInstance();
         this.recordWriter = RecordWriter.getInstance();
     }
 
     createLogger(customFields?: Map<string, any> | Object): Logger {
-
-        let logger = new Logger(this);
+        let logger = new Logger(this, this.context);
         // assign custom fields, if provided
         if (customFields) {
             logger.setCustomFields(customFields);
         }
         return logger;
     }
 
+    createContext() {
+        this.context = new Context();
+    }
+
```

Comment:
Call w/ @nicklas-dohrn:
- use flag in createLogger instead, e.g.,  createNewContext
- return bool when setting a context property to indicate success